version: '3.8'

services:
  proxy:
    image: traefik:v2.2
    restart: unless-stopped
    ports:
      - 80:80
      - 443:443
    networks:
      - proxy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./compose/traefik/config:/etc/traefik/config
      - ./compose/traefik/acme.json:/etc/traefik/acme.json
    command: 
      - --providers.docker=true
      - --providers.file.directory=/etc/traefik/config
      - --api.dashboard=true
      - --log.level=WARN
      - --entrypoints.web.address=:80
      - --entrypoints.websecure.address=:443
      - --certificatesresolvers.letsencrypt.acme.email=${EMAIL}
      - --certificatesresolvers.letsencrypt.acme.tlschallenge=true
      - --certificatesresolvers.letsencrypt.acme.storage=/etc/traefik/acme.json
      # Uncomment following line to test letsencrypt certificate issuing.
      # - --certificatesresolvers.letsencrypt.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory

    labels:
      - "traefik.http.routers.dashboard.rule=Host(`localhost`) && (PathPrefix(`/api`) || PathPrefix(`/dashboard`))"
      - "traefik.http.routers.dashboard.service=api@internal"

networks: 
  proxy:
    name: proxy_default
